src/plugins/controls/public/react_controls/control_group/get_control_group_factory.tsx

```diff
@@ -162,25 +163,27 @@ export const getControlGroupEmbeddableFactory = (services: {
           i18n.translate('controls.controlGroup.displayName', {
             defaultMessage: 'Controls',
           }),
-        openAddDataControlFlyout: () => {
-          /* openDataControlEditor({
+        openAddDataControlFlyout: (settings) => {
+          const { controlInputTransform } = settings ?? {
+            controlInputTransform: (state) => state,
+          };
+          openDataControlEditor({
             initialState: {
-              grow: DEFAULT_CONTROL_GROW,
-              width: DEFAULT_CONTROL_WIDTH,
-              dataViewId: parentApi.lastUsedDataViewId.getValue(),
+              grow: api.grow.getValue(),
+              width: api.width.getValue(),
             },
             onSave: ({ type: controlType, state: initialState }) => {
-              controlsManager.api.addNewPanel({
+              api.addNewPanel({
                 panelType: controlType,
-                initialState,
+                initialState: controlInputTransform!(
+                  initialState as Partial<ControlGroupSerializedState>,
```

Comment:
Shouldn't this be `Partial<DefaultControlState>` since the initialState is the control's initial state?

Reply:
Definitely a bad cast here, woops! Technically it should be `Partial<ControlInput>` because of how `ControlInputTransform` is defined in `src/plugins/controls/common/types.ts`. But since we don't want to touch any of the stuff that is shared between React controls and legacy, perhaps we just cast to `DataControlEditorState`? This satisfies `Partial<ControlInput>`

Reply:
sounds good. We can always iterator on this